### ISSUE TITLE
DisallowMultipleAssignments false positive when assigning to variable property of an object in an array

### ISSUE THREAD

**Issue by GaryJones** [PHPCSStandards/PHP_CodeSniffer#598]
## Describe the bug

False positive on a complex statement that suggests the assignment is not the first block of code.

Try assigning a value to a property of an object contained in an (indexed) array.

### Code sample
```php
<?php

$filtered_results[ $i ]->$field = $result;
```

### Custom ruleset
Can be reproduced with the `WordPress-Extra` or `Squiz` rulesets, or any combination using the `Squiz.PHP.DisallowMultipleAssignments` rule.

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php --standard=Squiz --sniffs=Squiz.PHP.DisallowMultipleAssignments -s` or similar
3. See error message displayed
```text
3 | ERROR | Assignments must be the first block of code on a line (Squiz.PHP.DisallowMultipleAssignments.Found)
```

## Expected behavior

No violation reported. 

## Versions (please complete the following information)

|                                       |                                                                                                           |
|--------------------------|----------------------------------------------------------------------  |
| Operating System           | MacOS 14.6.1                                                       |
| PHP version                    | 7.4                                                                                     |
| PHP_CodeSniffer version | 3.9.2                                                                            |
| Standard                         | Squiz                                                      |
| Install type                      | Composer |

## Additional context

None of the [tests](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/6fbbc1078094d905f0773421f13830744a144d1e/src/Standards/Squiz/Tests/PHP/DisallowMultipleAssignmentsUnitTest.1.inc) seem to have an assignment to a property of an object in an array.

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@GaryJones Thanks for reporting this. I can confirm the issue, which appears to be caused by the variable variable, i.e. the variable for the property name:

```php
$filtered_results[] = $result;
$filtered_results[ $i ] = $result;

$filtered_results->field = $result;
$filtered_results->$field = $result;

$filtered_results[ $i ]->field = $result;
$filtered_results[ $i ]->$field = $result; // <- This is the bug, all other test cases in this code sample are handled correctly.
```

**Comment by jrfnl** [comment]
@GaryJones PR #610 should fix this. Testing appreciated.
